packages/block-library/src/navigation/edit/index.js

```diff
@@ -485,6 +486,25 @@ function Navigation( {
 		{ open: overlayMenuPreview }
 	);
 
+	const submenuAccessibilityNotice =
+		! showSubmenuIcon && ! openSubmenusOnClick
+			? __(
+					'The current menu options offer reduced accessibility for users and are not recommended. '
+			  ) +
+			  __(
+					'Enabling either "Open on Click" or "Show arrow" offers enhanced accessibility by allowing keyboard users to browse submenus selectively.'
+			  )
+			: '';
+
+	useEffect( () => {
+		if ( submenuAccessibilityNotice )
```

Comment:
```suggestion
		if ( ! submenuAccessibilityNotice ) {
			return;
		}
```

Reply:
Done in https://github.com/WordPress/gutenberg/pull/52251/commits/5e15a923599b353eb8f4fbb5b6d2dedec148189b